src/plugins/discover/public/kibana_services.ts

```diff
@@ -76,4 +77,4 @@ export {
   ISearchSource,
   EsQuerySortValue,
   SortDirection,
-} from '../../../../../plugins/data/public';
+} from '../../data/public';
```

Comment:
I think re-exporting from other KP plugins/bundles was causing some problems a few month back. I remember an issue with `kibana_react` re-exporting things from `core`. But maybe that was only with concrete types. @spalger is there any issue with that?

Reply:
Since this code exists for a long, I didn't notice any problems with that.
If the issue you mentioned exists, I think we could create a separate fix for it and let this PR go further. 🙂

Reply:
I don't think there's anything wrong with this, I recommend this approach in many cases.